src/dust/dust.cpp

```diff
@@ -320,6 +457,344 @@ TaskStatus FluxSource(MeshData<Real> *md, const Real dt) {
+          for (int n = 0; n < nspecies; ++n) {
+            Real &dens_d = vmesh(b, dust::cons::density(n), k, j, i);
+            lsum += dens_d * vol00;
+          }
+          lmax1 = std::max(lmax1, nCalls(k, j, i));
+          for (int n = nspecies - 1; n >= 0; --n) {
+            Real &dens_d = vmesh(b, dust::cons::density(n), k, j, i);
+            if (dens_d > dfloor) {
+              lmax2 = std::max(lmax2, n);
+              break;
+            }
+          }
+        },
+        Kokkos::Sum<Real>(sumd0), Kokkos::Max<int>(maxCalls), Kokkos::Max<int>(maxSize));
+    sumd1 += sumd0;
+  } // loop of blocks
+
+#ifdef COAG_DEBUG
+  auto floor_h =
+      Kokkos::create_mirror_view_and_copy(parthenon::HostMemSpace(), floor_tmp);
+  for (int i = 0; i <= 0; i++) {
+    std::cout << "i=" << i << " ";
+    for (int n = 0; n < 20; n++) {
+      std::cout << floor_h(i, n, 0) << " ";
+    }
+    std::cout << std::endl;
+
+    std::cout << "i=" << i << " ";
+    for (int n = 0; n < 20; n++) {
+      std::cout << floor_h(i, n, 1) << " ";
+    }
+    std::cout << std::endl;
+  }
+#endif
+
+#ifdef MPI_PARALLEL
+  // over all processors
+  MPI_Reduce(MPI_IN_PLACE, &maxCalls, 1, MPI_INT, MPI_MAX, 0, MPI_COMM_WORLD);
+  MPI_Reduce(MPI_IN_PLACE, &maxSize, 1, MPI_INT, MPI_MAX, 0, MPI_COMM_WORLD);
+  MPI_Reduce(MPI_IN_PLACE, &sumd1, 1, MPI_PARTHENON_REAL, MPI_SUM, 0, MPI_COMM_WORLD);
+#endif // MPI_PARALLEL
+
+  if (parthenon::Globals::my_rank == 0) {
+    std::cout << "maxcalls,size= " << time << " " << maxCalls << " " << maxSize << " "
+              << maxSize0 << " " << time0 * dt << std::endl
+              << "total massd =" << time << " " << massd << " " << sumd1 << std::endl;
+  }
+
+  return TaskStatus::complete;
+}
+
+//----------------------------------------------------------------------------------------
+//! \fn  TaskCollection Dust::OperatorSplitDust
+//! \brief dust operator split task collection
+template <Coordinates GEOM>
+TaskCollection OperatorSplitDust(Mesh *pm, parthenon::SimTime &tm, const Real dt) {
+  TaskCollection tc;
+  auto &coag_pkg = pm->packages.Get("coagulation");
+  auto *dtCoag = coag_pkg->MutableParam<Real>("dtCoag");
+  int nstep1Coag = coag_pkg->template Param<int>("nstep1Coag");
+  if (tm.ncycle == 0) {
+    *dtCoag = 0.0;
+  }
+
+  *dtCoag += dt;
+  if ((tm.ncycle + 1) % nstep1Coag != 0) {
+    return tc;
+  }
+
+  const Real time_local = tm.time + dt - (*dtCoag);
+  const Real dt_local = (*dtCoag);
+  if (parthenon::Globals::my_rank == 0) {
+    std::cout << "coagulation at: time,dt,cycle=" << time_local << " " << dt_local << " "
+              << tm.ncycle << std::endl;
+  }
+
+  TaskID none(0);
+
+  // Assemble tasks
+  auto &coag = coag_pkg->template Param<Dust::Coagulation::CoagParams>("coag_pars");
+  const int nspecies = coag.nm;
+  std::vector<std::string> dust_var_names;
+  for (int n = 0; n < nspecies; n++) {
+    dust_var_names.push_back(dust::prim::density::name() + '_' + std::to_string(n));
```

Comment:
Do we really have to do this? I'd like to look at how to clean this up.

Reply:
If you have simple way to do it, please let me know

Reply:
Why do you need the names like this? I can't think of a reason why.